paychmgr/simple.go

```diff
@@ -120,8 +129,10 @@ func (pm *Manager) GetPaych(ctx context.Context, from, to address.Address, ensur
 	}
 	if ch != address.Undef {
 		// TODO: Track available funds
-		return ch, cid.Undef, pm.addFunds(ctx, ch, from, ensureFree)
+		mcid, err = pm.addFunds(ctx, ch, from, ensureFree)
+	} else {
+		mcid, err = pm.createPaych(ctx, from, to, ensureFree)
 	}
```

Comment:
Store lock could stay locked on error in `addFunds` / `createPaych`, but I think this is sufficient to fix that

```suggestion
	}

	if err != nil {
		pm.store.lk.Unlock()
	}
```

Reply:
good catch @shannonwells can you fix -- we want to make sure we unlock if the go routine is not getting kicked off

Reply:
I don't see `addFunds` taking a lock.

Reply:
it's called by GetPayCh which does

Reply:
overall some of our locking for all this is a bit unintuitive and we should probably address more permaneantly in a seperate PR -- there's a whole todo about tracking the payment channel even before it has an address but it's a bit tricky to think about what that looks like.

Reply:
addFunds calls waitForAddFunds, which accesses pm.state

Reply:
@magik6k there was another missed unlock at line 128.